From 82c2f059bc7c491c8c4c728408eb913bfc4a6f0a Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Wed, 6 Jun 2018 22:37:30 +0800
Subject: [PATCH] add soup_uri_to_string_with_password

The existed soup_uri_to_string does not have password, add a
function to support it.

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 libsoup/soup-uri.c | 21 +++++++++++++++++++++
 libsoup/soup-uri.h |  5 +++++
 2 files changed, 26 insertions(+)

diff --git a/libsoup/soup-uri.c b/libsoup/soup-uri.c
index 3eafd87..8421f1c 100644
--- a/libsoup/soup-uri.c
+++ b/libsoup/soup-uri.c
@@ -619,6 +619,27 @@ soup_uri_to_string (SoupURI *uri, gboolean just_path_and_query)
 }
 
 /**
+ * soup_uri_to_string_with_password:
+ * @uri: a #SoupURI
+ * @just_path_and_query: if %TRUE, output just the path and query portions
+ *
+ * Returns a string representing @uri.
+ *
+ * If @just_path_and_query is %TRUE, this concatenates the path and query
+ * together. That is, it constructs the string that would be needed in
+ * the Request-Line of an HTTP request for @uri.
+ *
+ * Note that the output will contain a password, if @uri does.
+ *
+ * Return value: a string representing @uri, which the caller must free.
+ **/
+char *
+soup_uri_to_string_with_password (SoupURI *uri, gboolean just_path_and_query)
+{
+	return soup_uri_to_string_internal (uri, just_path_and_query, TRUE, FALSE);
+}
+
+/**
  * soup_uri_copy:
  * @uri: a #SoupURI
  *
diff --git a/libsoup/soup-uri.h b/libsoup/soup-uri.h
index b9360c6..af702d3 100644
--- a/libsoup/soup-uri.h
+++ b/libsoup/soup-uri.h
@@ -57,6 +57,11 @@ char   	   *soup_uri_to_string             (SoupURI    *uri,
 					    gboolean    just_path_and_query);
 
 SOUP_AVAILABLE_IN_2_4
+char   	   *soup_uri_to_string_with_password             (SoupURI    *uri,
+					    gboolean    just_path_and_query);
+
+
+SOUP_AVAILABLE_IN_2_4
 SoupURI	   *soup_uri_copy                  (SoupURI    *uri);
 
 SOUP_AVAILABLE_IN_2_4
-- 
2.7.4

